kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/ContainerDescriptor.java

```diff
@@ -37,9 +37,9 @@
     private List<String> containerEnvVars;
     private List<String> containerDevices;
     private Map<String, String> containerVolumes;
-    private Boolean containerPrivilaged;
+    private Boolean containerPrivileged;
     private ContainerStates containerState = ContainerStates.STOPPING;
```

Comment:
Keeping the containerState field mutable can be error prone, it requires a lot of internal housekeeping and it can become inconsistent with system state.
What if we just remove it from here and add a method in DockerService that requests the container state to the container daemon given a container id?
Making ContainerDescriptor immutable would also be a nice imporvement.